src/app/stream-deck/actions.ts

```diff
@@ -0,0 +1,366 @@
+  });
+}
+
+// click on the nav link to open a specific page
+function goToPage(path: string) {
+  const [menuItem] = document.querySelectorAll(`a[href$="/${path}"]`);
+  const event = new MouseEvent('click', { bubbles: true, cancelable: true });
+  menuItem?.dispatchEvent(event);
+}
+
+// handle actions coming from the stream deck instance
+export function handleStreamDeckMessage(msg: string): ThunkResult {
+  return async (dispatch, getState) => {
+    const state = getState();
+
+    const data = checkAuthorization(msg, state.streamDeck, streamDeckWebSocket);
+
+    // this is not an encrypted msg or a challenge string
+    if (!data) {
+      return;
+    }
+
+    const currentStore = currentStoreSelector(state);
+
+    if (!currentStore) {
+      return;
+    }
+
+    switch (data.action) {
+      case 'refresh':
+        return refresh();
+      case 'search': {
+        goToPage(data.args.page || 'inventory');
+        setTimeout(() => dispatch(setSearchQuery(data.args.search, true)), 500);
+        return;
+      }
+      case 'randomize': {
+        const allItems = allItemsSelector(state);
+        const loadout = randomLoadout(
+          currentStore,
+          allItems,
+          data.args.weaponsOnly ? (i) => i.bucket?.sort === 'Weapons' : () => true
+        );
+        loadout && (await dispatch(applyLoadout(currentStore, loadout, { allowUndo: true })));
+        return;
+      }
+      case 'collectPostmaster': {
+        return dispatch(pullFromPostmaster(currentStore));
+      }
+      case 'farmingMode': {
+        if (state.farming.storeId) {
+          return dispatch(stopFarming());
+        } else {
+          return dispatch(startFarming(currentStore?.id));
+        }
+      }
+      case 'maxPower': {
+        const allItems = allItemsSelector(state);
+        const loadout = maxLightLoadout(allItems, currentStore);
+        return dispatch(applyLoadout(currentStore, loadout, { allowUndo: true }));
+      }
+      case 'selection': {
+        const selectionType = data.args.selection;
+        dispatch(setSearchQuery(''));
+        dispatch(streamDeckWaitSelection(selectionType));
+
+        // open the related page
+        goToPage(selectionType === 'loadout' ? 'loadouts' : 'inventory');
+
+        // show the notification
+        setTimeout(
+          () =>
+            showSelectionNotification(state, selectionType, () => {
+              dispatch(streamDeckClearSelection());
+            }),
+          500
+        );
+        return;
+      }
+      case 'loadout': {
+        const loadouts = loadoutsSelector(state);
+        const store = storesSelector(state).find((it) => it.id === data.args.character);
+        const loadout = loadouts.find((it) => it.id === data.args.loadout);
+        if (store && loadout) {
+          return dispatch(applyLoadout(store, loadout, { allowUndo: true }));
+        }
+        return;
+      }
+      case 'shareLoadout': {
```

Comment:
This one I don't quite follow - why would you share a loadout to the streamdeck?

Reply:
on long press on the Stream Deck button I ask the loadout shareable link to DIM and then it is copied to the clipboard through Node.js due to focus issue in the browser (when triggered outside the browser context)

Reply:
What I don't follow is why someone would want to do this?